packages/yarnpkg-core/sources/httpUtils.ts

```diff
@@ -100,31 +122,18 @@ export async function request(target: string, body: Body, {configuration, header
   const socketTimeout = configuration.get(`httpTimeout`);
   const retry = configuration.get(`httpRetry`);
   const rejectUnauthorized = configuration.get(`enableStrictSsl`);
-  const globalCaFilePath = configuration.get(`caFilePath`);
+  const caFilePath = networkConfig?.config.get(`caFilePath`) ?? configuration.get(`caFilePath`);
 
   const {default: got} = await import(`got`);
 
-  const extraHttpsOptions: HTTPSOptions = {};
-
-  for (const [glob, scopeConfig] of configuration.get(`networkSettings`)) {
-    if (micromatch.isMatch(url.hostname, glob)) {
-      extraHttpsOptions.certificateAuthority = await getCachedCertificate(scopeConfig.get(`caFilePath`));
```

Comment:
There was a bug here where it tried to read a file of `null` when the config looked like
```yml
networkSettings:
  "yarnpkg.com":
    enableNetwork: true
  "*":
    enableNetwork: false
```

cc @andreialecu